### PR TITLE
ALBS-32: Groups of actions for pes.almalinux

### PR DESCRIPTION
**Pull request by soksanichenko**
- Some additional fixes to migration from SQLite to PostgreSQL
- The separate dockerfile for container with PostgreSQL
- Full implementation groups of actions

### CODE REVIEW

File: ci/ansible/roles/deploy/tasks/migrate_sqlite_to_postgresql.yml [hunk 1]
```diff
@@ -5,23 +5,19 @@
...
-  community.docker.docker_container_exec:
-    container: "{{ container_env_prefix }}_postgres"
-    command: 'pgloader /migration_scenario'
-  when: db_is_migrated is defined and not db_is_migrated.stat.exists
-
-- name: Clear scenario of migration
-  file:
-    path: "{{ config_root }}/migration_scenario"
-    state: absent
+    - name: Run pqloader
```
Comment (Korulag) [inline ci/ansible/roles/deploy/tasks/migrate_sqlite_to_postgresql.yml]: Typo: `pqloader` -> `pgloader`

File: src/backend/api/handlers.py [hunk 2]
```diff
@@ -4,29 +4,57 @@
 from itertools import zip_longest
 from typing import (
     List,
-    Optional, Union, Tuple, Dict,
+    Optional,
+    Tuple,
```
Comment (gost-serb) [inline src/backend/api/handlers.py]: Starting from 3.9 it's preferred to use built-in types (like `list`, `tuple`...) over the ones from typing module

File: src/backend/api/handlers.py [hunk 3]
```diff
@@ -345,6 +417,7 @@ def add_or_edit_action_handler(
 
 def get_actions_handler(
         page: int,
+        group_id: int,
         page_size: int = PAGE_SIZE,
 ) -> Tuple[List[ActionData], Page]:
```
Comment (gost-serb) [inline src/backend/api/handlers.py]:
```suggestion
) -> tuple[list[ActionData], Page]:
```

File: src/backend/api/handlers.py [hunk 4]
```diff
@@ -458,9 +539,10 @@ def get_groups_of_actions_handler(
 ) -> Tuple[List[GroupActionsData], Page]:
```
Comment (gost-serb) [inline src/backend/api/handlers.py]:
```suggestion
) -> tuple[list[GroupActionsData], Page]:
```

File: src/backend/api/handlers.py [hunk 5]
```diff
@@ -230,12 +298,12 @@ def add_or_edit_group_of_actions_handler(
...
+            Group.create_from_dataclass(
                 session=db_session,
                 group_actions_data=group_actions_data,
             )
         else:
-            GroupActions.update_from_dataclass(
+            Group.update_from_dataclass(
                 session=db_session,
                 group_actions_data=group_actions_data,
             )
```
Comment (gost-serb) [inline src/backend/api/handlers.py]:
```python
            method = Group.create_from_class if is_new else Group.update_from_dataclass
            method(
                session=db_session,
                group_actions_data=group_actions_data,
            )
```

File: src/backend/db/data_models.py [hunk 6]
```diff
@@ -150,12 +151,14 @@ class GroupActionsData(BaseData):
...
     __skip_fields__ = (
         'id',
+        'actions_ids',
     )
 
     id: int = None
     name: str = None
     description: str = None
     github_org: GitHubOrgData = None
+    actions_ids: List[int] = None
```
Comment (gost-serb) [inline src/backend/db/data_models.py]:
```suggestion
    actions_ids: list[int] = None
```

File: src/backend/common/forms.py [hunk 7]
```diff
@@ -131,14 +164,23 @@ def to_dataclass(self) -> GroupActionsData:
             name=self.name.data,
             description=self.description.data,
             github_org=GitHubOrgData(
-                name=self.org.data,
-            )
+                github_id=self.org.data,
+            ),
+            actions_ids=[
+                int(action_id.strip()) for action_id in
+                self.actions_ids.data.split(',')
```
Comment (gost-serb) [inline src/backend/common/forms.py]:
```suggestion
                int(action_id.strip())
                for action_id in self.actions_ids.data.split(',')
```

This formatting makes it easier to see where is the result part and where is the loop itself

File: src/backend/db/data_models.py [hunk 8]
```diff
@@ -301,24 +306,27 @@ class ActionData(BaseData):
...
     is_approved: Optional[bool] = False
-    github_org: str = None
+    github_org: GitHubOrgData = None
     source_release: ReleaseData = field(default_factory=ReleaseData)
     target_release: ReleaseData = field(default_factory=ReleaseData)
     action: ActionType = None
     in_package_set: List[PackageData] = field(default_factory=list)
     out_package_set: List[PackageData] = field(default_factory=list)
     arches: List[str] = field(default_factory=list)
+    groups: List[GroupActionsData] = field(default_factory=list)
```
Comment (gost-serb) [inline src/backend/db/data_models.py]: Here and in other places `typing.List` is deprecated ([PEP-585](https://peps.python.org/pep-0585/)) starting from 3.9 (if I'm not mistaken) and should be replaced with built-in `list`

**Review by anfimovdm — APPROVED** [review]
lgtm

**Review by maccelf — APPROVED** [review]
LGTM

File: src/backend/api/handlers.py [hunk 9]
```diff
@@ -182,16 +245,16 @@ def bulk_upload_handler(json_dict: dict) -> None:
         action['action'] = ActionType.get_name(action['action'])
         action.pop('id', None)
         for root_key in (
-            'initial_release',
-            'release',
-            'in_packageset',
-            'out_packageset',
+                'initial_release',
```
Comment (Korulag) [inline src/backend/api/handlers.py]: Why additional indent?

**Review by anfimovdm — APPROVED** [review]
lgtm

**Pull request merged**